networkx/algorithms/tests/test_lowest_common_ancestors.py

```diff
@@ -311,3 +313,121 @@ def test_lowest_common_ancestor2(self):
         G = nx.DiGraph()
         G.add_node(3)
         assert nx.lowest_common_ancestor(G, 3, 3) == 3
+
+
+class TestNaiveLCA:
+    @classmethod
+    def setup_class(cls):
+        cls.DG = nx.DiGraph()
+        cls.DG.add_nodes_from(range(5))
+        cls.DG.add_edges_from([(1, 0), (2, 0), (3, 2), (4, 1), (4, 3)])
+
+        cls.root_distance = nx.shortest_path_length(cls.DG, source=4)
+
+        cls.gold = {
+            (0, 0): 0,
+            (0, 1): 1,
+            (0, 2): 2,
+            (0, 3): 3,
+            (0, 4): 4,
+            (1, 1): 1,
+            (1, 2): 4,
+            (1, 3): 4,
+            (1, 4): 4,
+            (2, 2): 2,
+            (2, 3): 3,
+            (2, 4): 4,
+            (3, 3): 3,
+            (3, 4): 4,
+            (4, 4): 4,
+        }
+
+    def assert_lca_dicts_same(self, d1, d2, G=None):
+        """Checks if d1 and d2 contain the same pairs and
+        have a node at the same distance from root for each.
+        If G is None use self.DG."""
+        if G is None:
+            G = self.DG
+            root_distance = self.root_distance
+        else:
+            roots = [n for n, deg in G.in_degree if deg == 0]
+            assert len(roots) == 1
+            root_distance = nx.shortest_path_length(G, source=roots[0])
+
+        for a, b in ((min(pair), max(pair)) for pair in chain(d1, d2)):
+            assert (
+                root_distance[get_pair(d1, a, b)] == root_distance[get_pair(d2, a, b)]
+            )
```

Comment:
If I'm not missing anything, it looks like the `G=None` arg isn't used in any of the following tests. If so, then this check could be made easier to understand by removing the unused kwarg and associated scaffolding, e.g.

```suggestion
    def assert_lca_dicts_same(self, d1, d2):
        """Checks if d1 and d2 contain the same pairs and
        have a node at the same distance from root for each.
        """
        for a, b in ((min(pair), max(pair)) for pair in chain(d1, d2)):
            assert (
                self.root_distance[get_pair(d1, a, b)] ==
                self.root_distance[get_pair(d2, a, b)]
            )
```

Reply:
Ah - so I just looked at the rest of the file and realized that this method (along with most of the other patterns, e.g. the test method names) were taken from the existing test classes, which makes total sense. I think there's actually a lot to clean up in the *existing* tests, so let's not worry about this here... I will create a followup issue so that we can handle this separately and not block this PR.

I'm going to switch my previous comment to an approval, sorry for the noise @dtekinoglu !